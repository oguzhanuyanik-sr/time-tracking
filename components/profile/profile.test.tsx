import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Profile from './index';

describe('avatar image', () => {
  test('image: is in document? is visible?', () => {
    render(<Profile />);

    const element = screen.getByRole('img');

    expect(element).toBeInTheDocument();
    expect(element).toBeVisible();
  });

  test('image: is alt correct?', () => {
    render(<Profile />);

    const element = screen.getByAltText('Avatar');

    expect(element).toHaveAttribute('width', '78');
    expect(element).toHaveAttribute('height', '78');
  });

  test('image: is src correct?', () => {
    render(<Profile />);

    const element = screen.getByAltText('Avatar');

    expect(element).toHaveAttribute(
      'src',
      expect.stringMatching(/image-jeremy.png/i)
    );
  });
});

describe('titles', () => {
  test('subtitle: is in document? is visible? is text correct?', () => {
    render(<Profile />);

    const element = screen.getByText('Report for');

    expect(element).toBeInTheDocument();
    expect(element).toBeVisible();
  });

  test('title: is in document? is text correct?', () => {
    render(<Profile />);

    const element = screen.getByRole('heading', {
      level: 1,
      name: 'Jeremy Robson',
    });

    expect(element).toBeInTheDocument();
  });
});

describe('nav', () => {
  test('list: is length correct?', () => {
    render(<Profile />);

    const element = screen.getByRole('list');

    expect(element.childNodes.length).toBe(3);
  });
});

test('snapshot', () => {
  const tree = renderer.create(<Profile />).toJSON();

  expect(tree).toMatchSnapshot();
});
