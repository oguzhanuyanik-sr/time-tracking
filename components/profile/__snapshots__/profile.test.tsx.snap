// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="componentWrapper"
>
  <section
    className="profileWrapper"
  >
    <div>
      <img
        alt="Avatar"
        className="profileAvatar"
        data-nimg="1"
        decoding="async"
        height={78}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        sizes="100%"
        src="/_next/image?url=%2Fimages%2Fimage-jeremy.png&w=3840&q=75"
        srcSet="/_next/image?url=%2Fimages%2Fimage-jeremy.png&w=16&q=75 16w, /_next/image?url=%2Fimages%2Fimage-jeremy.png&w=32&q=75 32w, /_next/image?url=%2Fimages%2Fimage-jeremy.png&w=48&q=75 48w, /_next/image?url=%2Fimages%2Fimage-jeremy.png&w=64&q=75 64w, /_next/image?url=%2Fimages%2Fimage-jeremy.png&w=96&q=75 96w, /_next/image?url=%2Fimages%2Fimage-jeremy.png&w=128&q=75 128w, /_next/image?url=%2Fimages%2Fimage-jeremy.png&w=256&q=75 256w, /_next/image?url=%2Fimages%2Fimage-jeremy.png&w=384&q=75 384w, /_next/image?url=%2Fimages%2Fimage-jeremy.png&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fimage-jeremy.png&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fimage-jeremy.png&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fimage-jeremy.png&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fimage-jeremy.png&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fimage-jeremy.png&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fimage-jeremy.png&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fimage-jeremy.png&w=3840&q=75 3840w"
        style={
          {
            "color": "transparent",
          }
        }
        width={78}
      />
    </div>
    <div>
      <span
        className="profileSubtitle"
      >
        Report for
      </span>
      <h1
        className="profileTitle"
      >
        Jeremy Robson
      </h1>
    </div>
  </section>
  <section>
    <ul
      className="navWrapper"
    >
      <li
        className="navButton "
        onClick={[Function]}
      >
        daily
      </li>
      <li
        className="navButton "
        onClick={[Function]}
      >
        weekly
      </li>
      <li
        className="navButton "
        onClick={[Function]}
      >
        monthly
      </li>
    </ul>
  </section>
</div>
`;
